### PR TITLE
Added unsloth training script & newest version of data generation + benchmarking

### PR DESCRIPTION
**Pull request by dendelyne**

### CODE REVIEW

File: data_and_training/benchmarking/format_text.py
Comment (isspek) [inline data_and_training/benchmarking/format_text.py:1]: is it from the repo for data generation? are they synchroized?

File: data_and_training/unsloth-training/train_unsloth.py
Comment (isspek) [inline data_and_training/unsloth-training/train_unsloth.py:1]: I would add loading function to test which is transformer-based approach to load the model. so we can make sure that uploaded model has no issue locally.

File: data_and_training/unsloth-training/train_unsloth.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    # model.push_to_hub_merged(f'{output_name}_merged_4bit', tokenizer, save_method="merged_4bit_forced", token=hf_token)
+
+    model.save_pretrained(f'models/{output_name}_local')  # Local saving
+    tokenizer.save_pretrained(f'models/{output_name}_local')
+
+    model.push_to_hub(f'{output_name}_lora', token=hf_token)  # Online saving
+    tokenizer.push_to_hub(f'{output_name}_lora', token=hf_token)  # Online saving
+
+    # model.save_pretrained_merged(f'{output_name}_lora', tokenizer, save_method = "lora", token=hf_token)
+    # model.push_to_hub_merged(f'{output_name}_lora', tokenizer, save_method = "lora", token=hf_token)
```
Comment (isspek) [inline data_and_training/unsloth-training/train_unsloth.py:153]: I wonder if this was the issue of why we can not load the model :/ maybe we use these methods too.

**Comment by dendelyne** [comment]
Instead of directly uploading the scripts to this repo, submodules will be used to point to the two external repos:
* https://github.com/dw-innovation/kid2-spot-datageneration
* https://github.com/dw-innovation/unsloth-training

This PR will therefore be closed.

**Pull request closed without merging**